Incremental improvements to TMO landing page

georgf opened the pull request:
Following up on initial insights from user interviews and a planning meeting, i want to put out some short-term & cheap fixes to the current landing page.
This would help us with highlighting the page as the "go to" page for Telemetry & data things in Austin.

We definitely want to do a proper re-design later, but that doesn't seem feasible in the short term.
I played around with some smaller changes, you can see them live [here](https://georgf.github.io/telemetry-dashboard/).

To summarize:
- added a lock icon to LDAP protected pages
- moved the hardware report to main dashboards & reports.tmo out of it
- added a docs section, with links to docs.tmo, client docs & the probe explorer
- added a getting help section with links to our communication channels

I'm not happy with the layout changes, feedback appreciated. Specifically i can't add two `scrollable-dashboard-list` to one column as the second one gets squashed to the bottom, that's why i moved the `h3`s inside one list.

georgf committed:
Move Hardware Report to main dashboards

This dashboard is hard to find, but answers some important questions
for engineers and the broader user base.

georgf committed:
Make hardware report description more snappy

georgf committed:
Highlight & expand documentation links

georgf committed:
Fix column height to avoid scroll bars

georgf committed:
Add "getting help" section

georgf committed:
Minor rewording

georgf committed:
Make clear which items are LDAP protected

georgf committed:
Add experiments viewer dashboard

georgf committed:
Move reports.tmo due to column overflow

georgf commented:
Specific feedback i'm looking for:
- is that reasonable as short-term usability fixes?
- do i have the right items?
- is that ok-ish layout-wise for now?

chutten commented on index.html:
```diff
@@ -33,28 +30,45 @@ <h2>Telemetry dashboards</h2>
             </div>
             <div class="col-md-4">
-                <h3>Telemetry @ Twitter</h3>
+                <h3>What is new</h3>
```
bikeshed alert: Maybe prefer "What's new" as it is more commonly written with the contraction.

chutten reviewed (commented):
Happy to hear what my fellow reviewers have to say.

georgf committed:
Make the experiments viewer description more snappy

georgf committed:
Change "what is new" to "what's new"

georgf committed:
Change the page title to "Telemetry portal

fbertsch commented:
Looks good @georgf!

- Can the "Getting Help" and "Other Dashboards not be in a separate scroll window, but just scroll with the page?
- Why don't we just actually remove all of those old dashboards, no one has asked about them
- Mission Control should be on here somewhere
- Lock icon is a great solution!

georgf committed:
Add Shield docs

georgf committed:
Make columns non-scrollable

Make columns non-scrollable, instead make them grow with the page and
let people scroll the page.

georgf committed:
Remove non-existing dashboards

Remove non-existing dashboards, instead add an item that tells you to
file a bug for anything missing.

georgf committed:
Break up lists into separate dashboard-list elements

Now that we don’t use scrollable list anymore, we can just use separate
lists for the different use-cases.

georgf commented:
> - Can the "Getting Help" and "Other Dashboards not be in a separate scroll window, but just scroll with the page?
> - Why don't we just actually remove all of those old dashboards, no one has asked about them

I was wondering that too. I pushed both now and added a "missing anything? file a bug" item.
Chris, any issues you see with those changes?

> Mission Control should be on here somewhere

I didn't really understand if we officially want to point people to it yet.
@rafrombrc, should i add it to "other dashboards"? And with what link?

georgf committed:
Make clear that sql.tmo shows dashboards too

georgf committed:
Add Amplitude

georgf committed:
Use "tools" instead of "dashboards"

Not all things we link to are dashboards, lets use something more
generic.

fbertsch commented:
Just a few more nits:
- Not all of the titles have all main words capitalized (e.g. `Experiments viewer`)
- The `Missing Something` is grayed out
- We can probably remove the "Telemetry Wiki" link, everything has been migrated to `docs.tmo`

erikrose commented on index.html:
```diff
@@ -22,39 +22,58 @@
             </div>
             <div class="col-md-4">
-                <h3>Telemetry @ Twitter</h3>
+                <h3>What's new</h3>
```
What are the relative importances of the What's New Twitter stream and the links to dashboards? If What's New is least important, I'd move it to the far right column so it comes last in very narrow (or mobile) windows.

erikrose replied:
FWIW, the Twitter stream doesn't load in Nightly or Safari for me, though it does in Chrome. Could well be my various blocking extensions in Nightly. Not sure about Safari.

georgf replied:
It's probably blocking, for me they load.

georgf replied:
Re "whats new": This gets read by at least some people.
The "getting help" section and "other tools" are probably looked for more actively by people, so should (?) be fine on the right.

erikrose commented on index.html:
```diff
@@ -22,39 +22,58 @@
+                    <a class="list-group-item" href="https://georgf.github.io/fx-data-explorer/">
+                        Probe dictionary <div class="dashboard-description">Find what probes are submitted from Firefox.</div>
+                    </a>
+                    <a class="list-group-item" href="https://wiki.mozilla.org/Firefox/Shield">
```
Kudos that (as before) the entire rectangles of the list-group-items are clickable.

erikrose commented on index.html:
```diff
@@ -109,40 +150,8 @@ <h3>Other dashboards</h3>
-                        SlowSQL
-                        <div class="dashboard-description">Please <a href="https://github.com/mozilla/telemetry-dashboard/issues">Report a Bug</a> if you require this data.</div>
+                        Missing something?
+                        <div class="dashboard-description">Please <a href="https://github.com/mozilla/telemetry-dashboard/issues">Report a Bug</a> if you are missing a dashboard here.</div>
```
You've got 2 Report Bug links that point to the same place. Is the one up top necessary? I generally *do* look at the bottom for bug-report links.

georgf replied:
I thought it might be a friendly "next step" for people looking through "other tools" and not finding what they found.
On the other hand, the top-right link covers looking around generally when something is broken. Does the "getting help" section cover this now?

georgf replied:
I'm dropping this one, it's probably not adding anything and potentially hidden off-page, behind scroll.

erikrose commented on index.html:
```diff
@@ -22,39 +22,58 @@
             <form class="navbar-form inline pull-right">
                 <a href="https://wiki.mozilla.org/Telemetry" class="btn" title="Wiki" target="_blank">Telemetry Wiki</a>
```
Why is Telemetry Wiki special enough to merit its own top-of-the-page link? Would it do better under Documentation, or is it a mishmash of different things?

And a personal opinion: I dislike target="_blank" links, because it's not evident they're going to act differently, and I rarely desire that behavior.

erikrose replied:
Ah, I see Frank says we can remove it completely. Spiffy.

erikrose reviewed (commented):
My favorite thing about the new design is that it eliminates the scrolls within scrolls. Now I can use Page Down to navigate the page! :-D

georgf committed:
Remove wiki link

georgf committed:
Remove duplicated bug report link

This is probably triplicated.

georgf committed:
Use more consistent capitalisation

chutten commented on style/dashboards.css:
```diff
@@ -17,9 +17,13 @@
 }
+.dashboard-list .list-group-item {
+  border: none;
+  border-top: 1px solid #ddd;
```
If this is making the non-scrolling dashboard list like `.scrollable-dashboard-list`, we'll also need to grab the `:first-child` style that disables the topmost `border-top`

chutten commented on style/dashboards.css:
```diff
@@ -17,9 +17,13 @@
+  border-top: 1px solid #ddd;
+}
 .big-dashboard-list .list-group-item .dashboard-description { font-size: 18px; }
 .scrollable-dashboard-list {
```
You've removed the only two `.scrollable-dashboard-list` in the repo, so we can safely remove their styles.

georgf committed:
Remove unused .scrollable-dashboard-list styling

georgf committed:
Remove top border on first dashboard-list items

chutten reviewed (approved):
Looks good! Please squash these commits together for rebase and merge, and we'll get them in.
